RecoMuon/TrackerSeedGenerator/plugins/TSGForOIDNN.cc

```diff
@@ -0,0 +1,974 @@
+      delete graphDef_HB_;
+      delete graphDef_HLIP_;
+      delete graphDef_HLMuS_;
+    } else {
+      tensorflow::closeSession(tf_session_barrel_);
+      tensorflow::closeSession(tf_session_endcap_);
+      delete graphDef_barrel_;
+      delete graphDef_endcap_;
+    }
+  }
+}
+
+//
+// Produce seeds
+//
+void TSGForOIDNN::produce(edm::StreamID sid, edm::Event& iEvent, edm::EventSetup const& iEventSetup) const {
+  // Initialize variables
+  unsigned int numSeedsMade = 0;
+  unsigned int layerCount = 0;
+  unsigned int hitlessSeedsMadeIP = 0;
+  unsigned int hitlessSeedsMadeMuS = 0;
+  unsigned int hitSeedsMade = 0;
+  unsigned int hitDoubletSeedsMade = 0;
+
+  // Surface used to make a TSOS at the PCA to the beamline
+  Plane::PlanePointer dummyPlane = Plane::build(Plane::PositionType(), Plane::RotationType());
+
+  // Read ESHandles
+  edm::ESHandle<Chi2MeasurementEstimatorBase> estimatorH = iEventSetup.getHandle(t_estimatorH_);
+  edm::ESHandle<MagneticField> magfieldH = iEventSetup.getHandle(t_magfieldH_);
+  edm::ESHandle<Propagator> propagatorAlongH = iEventSetup.getHandle(t_propagatorAlongH_);
+  edm::ESHandle<Propagator> propagatorOppositeH = iEventSetup.getHandle(t_propagatorOppositeH_);
+  edm::ESHandle<TrackerGeometry> tmpTkGeometryH = iEventSetup.getHandle(t_tmpTkGeometryH_);
+  edm::ESHandle<GlobalTrackingGeometry> geometryH = iEventSetup.getHandle(t_geometryH_);
+  edm::ESHandle<NavigationSchool> navSchool = iEventSetup.getHandle(t_navSchool_);
+
+  edm::Handle<MeasurementTrackerEvent> measurementTrackerH;
+  iEvent.getByToken(measurementTrackerTag_, measurementTrackerH);
+
+  // Read L2 track collection
+  edm::Handle<reco::TrackCollection> l2TrackCol;
+  iEvent.getByToken(src_, l2TrackCol);
+
+  // The product
+  std::unique_ptr<std::vector<TrajectorySeed> > result(new std::vector<TrajectorySeed>());
+
+  // Get vector of Detector layers
+  std::vector<BarrelDetLayer const*> const& tob = measurementTrackerH->geometricSearchTracker()->tobLayers();
+  std::vector<ForwardDetLayer const*> const& tecPositive =
+      tmpTkGeometryH->isThere(GeomDetEnumerators::P2OTEC)
+          ? measurementTrackerH->geometricSearchTracker()->posTidLayers()
+          : measurementTrackerH->geometricSearchTracker()->posTecLayers();
+  std::vector<ForwardDetLayer const*> const& tecNegative =
+      tmpTkGeometryH->isThere(GeomDetEnumerators::P2OTEC)
+          ? measurementTrackerH->geometricSearchTracker()->negTidLayers()
+          : measurementTrackerH->geometricSearchTracker()->negTecLayers();
+
+  // Get suitable propagators
+  std::unique_ptr<Propagator> propagatorAlong = SetPropagationDirection(*propagatorAlongH, alongMomentum);
+  std::unique_ptr<Propagator> propagatorOpposite = SetPropagationDirection(*propagatorOppositeH, oppositeToMomentum);
+
+  // Stepping Helix Propagator for propogation from muon system to tracker
+  edm::ESHandle<Propagator> SHPOpposite = iEventSetup.getHandle(t_SHPOpposite_);
```

Comment:
```suggestion
  edm::ESHandle<Propagator> shpOpposite = iEventSetup.getHandle(t_SHPOpposite_);
```
start local variables and data members with lower case letter. Capitalization is reserved for class names.

Reply:
Done!